contrib/tools/config-docs-generator/generate-config-docs.sh

```diff
@@ -0,0 +1,145 @@
+OUTPUT_DIR="$PROJECT_ROOT/docs/generated"
+CARGO_TARGET_DIR="${CARGO_TARGET_DIR:-$PROJECT_ROOT/target}"
+TEMP_DIR="${TEMP_DIR:-$CARGO_TARGET_DIR/doc-generation}"
+
+# Binary paths - allow override via environment
+EXTRACT_DOCS_BIN="${EXTRACT_DOCS_BIN:-$CARGO_TARGET_DIR/release/extract-docs}"
+GENERATE_MARKDOWN_BIN="${GENERATE_MARKDOWN_BIN:-$CARGO_TARGET_DIR/release/generate-markdown}"
+
+# Template and mappings paths - allow override via environment
+TEMPLATE_PATH="${TEMPLATE_PATH:-$SCRIPT_DIR/templates/reference_template.md}"
+SECTION_MAPPINGS_PATH="${SECTION_MAPPINGS_PATH:-$SCRIPT_DIR/section_name_mappings.json}"
+
+# Check if binaries are pre-built (skip build step)
+SKIP_BUILD="${SKIP_BUILD:-false}"
+
+export CARGO_TARGET_DIR
+
+log_info() {
+    echo -e "${GREEN}[INFO]${NC} $1"
+}
+
+log_warn() {
+    echo -e "${YELLOW}[WARN]${NC} $1"
+}
+
+log_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+cleanup() {
+    if [[ -d "$TEMP_DIR" ]]; then
+        rm -rf "$TEMP_DIR"
+    fi
+}
+
+trap cleanup EXIT
+
+main() {
+    log_info "Starting config documentation generation..."
+
+    # Create necessary directories
+    mkdir -p "$OUTPUT_DIR"
+    mkdir -p "$TEMP_DIR"
+
+    cd "$PROJECT_ROOT"
+
+    # Step 1: Build the documentation generation tools
+    if [[ "$SKIP_BUILD" != "true" ]]; then
+        log_info "Building documentation generation tools..."
+        cargo build --package config-docs-generator --release
+    fi
+
+    # Step 2: Extract documentation from source code using rustdoc
+    log_info "Extracting configuration documentation using rustdoc..."
+    EXTRACTED_JSON="$TEMP_DIR/extracted-config-docs.json"
+    # List of specific Rust struct names to be documented
+    # NOTE: This variable must be manually updated if this list changes
+    # (e.g., new config structs are added or removed from the project)
+    TARGET_STRUCTS="BurnchainConfig,NodeConfig,MinerConfig,ConnectionOptionsFile,FeeEstimationConfigFile,EventObserverConfigFile,InitialBalanceFile"
+    "$EXTRACT_DOCS_BIN" \
+        --package stackslib \
+        --structs "$TARGET_STRUCTS" \
+        --output "$EXTRACTED_JSON"
+
+    # Step 3: Generate Markdown
+    log_info "Generating Markdown documentation..."
+    MARKDOWN_OUTPUT="$OUTPUT_DIR/configuration-reference.md"
+
+    # Call the command
+    "$GENERATE_MARKDOWN_BIN" --input "$EXTRACTED_JSON" --output "$MARKDOWN_OUTPUT" --template "$TEMPLATE_PATH" --section-name-mappings "$SECTION_MAPPINGS_PATH"
+
+    log_info "Documentation generation complete!"
+    log_info "Generated files:"
+    log_info "  - Configuration reference: $MARKDOWN_OUTPUT"
+    log_info "  - Intermediate JSON: $EXTRACTED_JSON"
+
+    # Verify output
+    if [[ -f "$MARKDOWN_OUTPUT" ]]; then
+        WORD_COUNT=$(wc -w < "$MARKDOWN_OUTPUT")
+        log_info "Generated Markdown contains $WORD_COUNT words"
+    else
+        log_error "Expected output file not found: $MARKDOWN_OUTPUT"
+        exit 1
```

Comment:
related to my previous comment about log_exit - you could move the 2 instances i see of log_error `exit 1` to the `log_error` function and save a few chars, unless there is a possibility the exit codes will be different based on the error.